query-service/src/integrationTest/java/org/hypertrace/core/query/service/htqueries/HTPinotQueriesTest.java

```diff
@@ -0,0 +1,344 @@
+        .withNetwork(network)
+        .withNetworkAliases("query-service")
+        .withEnv("ATTRIBUTE_SERVICE_HOST_CONFIG", attributeService.getHost())
+        .withEnv("ATTRIBUTE_SERVICE_PORT_CONFIG", attributeService.getMappedPort(9012).toString())
+        .withEnv("ZK_CONNECT_STR", "zookeeper:2181/hypertrace-views")
+        .withExposedPorts(8090)
+        .dependsOn(pinotServiceManager)
+        .dependsOn(attributeService)
+        .withStartupAttempts(CONTAINER_STARTUP_ATTEMPTS)
+        .waitingFor(Wait.forLogMessage(".*Started admin service on port: 8091.*", 1));
+    queryService.start();
+    queryService.followOutput(logConsumer);
+
+    List<String> topicsNames = List.of(
+        "enriched-structured-traces",
+        "raw-service-view-events",
+        "raw-trace-view-events",
+        "service-call-view-events",
+        "span-event-view",
+        "backend-entity-view-events");
+    bootstrapServers = kafkaZk.getBootstrapServers();
+    adminClient = AdminClient
+        .create(Map.of(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaZk.getBootstrapServers()));
+    List<NewTopic> topics = topicsNames.stream().map(v -> new NewTopic(v, 1, (short) 1))
+        .collect(Collectors.toList());
+    adminClient.createTopics(topics);
+
+    assertTrue(bootstrapConfig());
+    LOG.info("Bootstrap Complete");
+    assertTrue(generateData());
+    LOG.info("Generate Data Complete");
+
+    Map<String, Object> map = Maps.newHashMap();
+    map.put("host", queryService.getHost());
+    map.put("port", queryService.getMappedPort(8090));
+    QueryServiceConfig queryServiceConfig = new QueryServiceConfig(ConfigFactory.parseMap(map));
+    queryServiceClient = new QueryServiceClient(queryServiceConfig);
+  }
+
+  @AfterAll
+  public static void shutdown() {
+    LOG.info("Initiating shutdown");
+    queryService.stop();
+    attributeService.stop();
+    mongo.stop();
+    pinotServiceManager.stop();
+    kafkaZk.stop();
+    network.close();
+  }
+
+  private static boolean bootstrapConfig() throws Exception {
+    String resourcesPath =
+        Thread.currentThread()
+            .getContextClassLoader()
+            .getResource("config-bootstrapper")
+            .getPath();
+
+    // Since the clients to run Config commands are created internal to this code,
+    // we need to set the tenantId in the context so that it's propagated.
+    RequestContext requestContext = new RequestContext();
+    requestContext.add(RequestContextConstants.TENANT_ID_HEADER_KEY, "__default");
+    Context context = Context.current().withValue(RequestContext.CURRENT, requestContext);
+
+    withEnvironmentVariable(
+        "MONGO_PORT", mongo.getMappedPort(27017).toString())
+        .and("ATTRIBUTE_SERVICE_HOST_CONFIG", attributeService.getHost())
+        .and("ATTRIBUTE_SERVICE_PORT_CONFIG", attributeService.getMappedPort(9012).toString())
+        .execute(() -> context.run(
+            () -> ConfigBootstrapper.main(
```

Comment:
Can you also run `config-bootstrapper` task using the test container?

Reply:
I started with that. Since config-bootstrapper exits after finishing, bringing it up as a container is causing problems.

Reply:
What if you start after its dependent container's started?

Reply:
what if we take these entire tests at `hypertrace-service` level as for this service, it seems we need to bring a full-stack?

Reply:
> Can you also run config-bootstrapper task using the test container?

Done